src/context/WalletProvider/NativeWallet/components/NativeCreate.tsx

```diff
@@ -83,36 +84,40 @@ export const NativeCreate = () => {
   }, [history, isLegacyWallet, mixpanel, vault])
 
   useEffect(() => {
+    if (vault) return
     ;(async () => {
       try {
-        // If the vault is already passed from the legacy wallet flow, use it.
-        const vault = isLegacyWallet ? location.state.vault : await getVault()
+        const vault = location.state?.vault ?? (await getVault())
         setVault(vault)
       } catch (e) {
         console.error(e)
       }
     })()
-  }, [setVault, location.state?.vault, isLegacyWallet])
+  }, [setVault, vault, location.state?.vault])
 
   useEffect(() => {
     if (!vault) return
     ;(async () => {
       try {
+        revokerRef.current.revoke()
+        revokerRef.current = new (Revocable(class {}))()
+
+        const mnemonic = await vault.unwrap().get('#mnemonic')
```

Comment:
Just for readability and debugging purposes

Reply:
Confirmed this is just moved around, nothing new to see in terms of exposing things